gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -133,14 +134,26 @@ public static void setGCloudOptions(StorageOptions newStorageOptions) {
    * @see CloudStorageFileSystem#forBucket(String)
    */
   public CloudStorageFileSystemProvider() {
-    this(storageOptions);
+    this(futureStorageOptions);
   }
 
   CloudStorageFileSystemProvider(@Nullable StorageOptions gcsStorageOptions) {
-    if (gcsStorageOptions == null) {
+    this.storageOptions = gcsStorageOptions;
+
+  }
+
+  // Initialize this.storage, once. This may throw an exception if the environment variable
+  // GOOGLE_APPLICATION_CREDENTIALS is not set.
+  // We don't do this in the constructor because the ctor is called even when just enumerating
+  // available filesystems, and this should work without configuring gcloud.
```

Comment:
This removes the most important part of the comment, which is to explain why we are not doing things the natural way. How about this instead:

```
// Initialize this.storage, once. This may throw an exception if default authentication
// credentials are not available (hence not doing it in the ctor).
```